pkg/target/target.go

```diff
@@ -508,31 +512,61 @@ func Summary(targets []*Target) fleet.BundleSummary {
 	return bundleSummary
 }
 
-func processLabelValues(valuesMap map[string]interface{}, clusterLabels map[string]string) error {
-	prefix := "global.fleet.clusterLabels."
+func processValues(valuesMap map[string]interface{}, clusterLabels map[string]string, clusterAnnotations map[string]string) (err error) {
+	labelPrefix := "global.fleet.clusterLabels."
+	annotationPrefix := "global.fleet.clusterAnnotations."
+	scopedClusterLables := map[string]interface{}{
+		"global": map[string]interface{}{
+			"fleet": map[string]interface{}{
+				"clusterLabels":      clusterLabels,
+				"clusterAnnotations": clusterAnnotations,
+			},
+		},
+	}
 	for key, val := range valuesMap {
-		valStr, ok := val.(string)
-		if ok && strings.HasPrefix(valStr, prefix) {
-			label := strings.TrimPrefix(valStr, prefix)
-			labelVal, labelPresent := clusterLabels[label]
-			if labelPresent {
-				valuesMap[key] = labelVal
+		switch val.(type) {
+		case string:
+			valStr, _ := val.(string)
+			if strings.HasPrefix(valStr, labelPrefix) {
+				label := strings.TrimPrefix(valStr, labelPrefix)
+				labelVal, labelPresent := clusterLabels[label]
+				if labelPresent {
+					valuesMap[key] = labelVal
+				} else {
+					return fmt.Errorf("invalid_label_reference %s in key %s", valStr, key)
+				}
+			} else if strings.HasPrefix(valStr, annotationPrefix) {
+				annoation := strings.TrimPrefix(valStr, annotationPrefix)
+				annoationVal, annoationPresent := clusterLabels[annoation]
+				if annoationPresent {
+					valuesMap[key] = annoationVal
+				} else {
+					return fmt.Errorf("invalid_annotation_reference %s in key %s", valStr, key)
+				}
 			} else {
-				return fmt.Errorf("invalid_label_reference %s in key %s", valStr, key)
+				valuesTemplate, _ := template.New("values").Option("missingkey=error").Parse(valStr)
```

Comment:
If we are reusing the same template with multiple calls to `.Parse`, `template.New("values").Option("missingkey=error")` should be allocated once in the top of this file instead of allocated per override.

Reply:
we can put this into a variable like `tpl := template.New("values").Option("missingkey=error")` that is used in following calls